nautobot/circuits/models.py

```diff
@@ -9,17 +10,64 @@
 from nautobot.core.models.generics import BaseModel, OrganizationalModel, PrimaryModel
 from nautobot.utilities.utils import serialize_object
 from .choices import CircuitTerminationSideChoices
-from .querysets import CircuitQuerySet
 
 
 __all__ = (
     "Circuit",
     "CircuitTermination",
     "CircuitType",
     "Provider",
+    "ProviderNetwork",
 )
 
 
+@extras_features(
+    "custom_fields",
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "relationships",
+    "webhooks",
+)
+class ProviderNetwork(PrimaryModel):
+    name = models.CharField(max_length=100, unique=True)
```

Comment:
I think `unique=True` is incorrect here as below you're defining the `unique_together = ("provider", "name")` - it should be permissible to use the same name across different providers, I think?

Reply:
Since we are using this as a base for our slug, which needs to be unique, we may also want to enforce it here. 
If it is no problem for the slug handling, I am happy to remove it.

Reply:
`AutoSlugField` is supposed to automatically handle uniqueness at creation time - IIRC if you create Provider 1 Network "Alpha" (slug `alpha`) and Provider 2 Network "Alpha" (slug `alpha`) the latter will automatically become slug `alpha1` so as to have a unique slug - but by all means please try it out to confirm.

Reply:
It does not get automagically changed, but the creation gives a reasonable error.